mission/BouysTasksNjord/bouys_tasks_fsm.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+#Written by Sigurd von Brandis, Student
+
+import rospy
+import smach
+import smach_ros
+import math
+from bouys_tasks import Idle, Search, DetectedObjectsNavigation
+from nav_msgs.msg import Odometry
+from update_objects_data import DetectedObjectsData, UpdateDataNode  # custom message type for the combined data.
+from vortex_msgs.msg import DetectedObjectArray, DetectedObject
+
+
+class ManeuveringNavigationTasks:
+
+    def __init__(self):
+        rospy.init_node('Bouys_tasks_fsm')
+
+    def spin(self):
+
+        sm = smach.StateMachine(outcomes=['STOP'])
+
+        with sm:
+            smach.StateMachine.add('Idle',
+                                   Idle(),
+                                   transitions={
+                                       'detectedObjectsNavigation':
+                                       'DetectedObjectsNavigation',
+                                       'search': 'Search'
+                                   },
+                                   remapping={
+                                       'closest_object':
+                                       'closest_object',
+                                       'object_search_attempts':
+                                       'object_search_attempst'
+                                   })
+
+            smach.StateMachine.add('Search',
+                                   Search(self.data),
+                                   transitions={'idle': 'Idle'},
+                                   remapping={
+                                       'closest_object':
+                                       'closest_object',
+                                       'object_search_attempst':
+                                       'object_search_attempst'
+                                   })
+
+            smach.StateMachine.add('DetectedObjectsNavigation',
+                                   DetectedObjectsNavigation(),
+                                   transitions={'idle'},
+                                   remapping={
+                                       'closest_object':
+                                       'closest_object',
+                                       'object_search_attempst':
+                                       'object_search_attempst'
+                                   })
+
+        # Start the state machine introspection server
+        sis = smach_ros.IntrospectionServer('state_machine', sm, '/SM_ROOT')
+        sis.start()
+
+        while not rospy.is_shutdown():
+            # To be updated
```

Comment:
Remove comments if not needed, mark with TODO if future work, clean up otherwise